### ISSUE TITLE
twitter show <username>

### ISSUE THREAD

**Issue by devth** [devth/yetibot#564]
Show the top 10 tweets for `username`. Maybe include options to exclude retweets and replies. Related: #552

**Comment by dev-hartmann** [comment]
Hey there, I'm new to Clojure and would like to tackle this issue. I guess at some point I will be asking a few questions ;)

**Comment by devth** [comment]
Awesome, feel free to ask anytime!

**Comment by devth** [comment]
Here's a few suggestions to get started.

You'll want to add a subcommand regex after this line:
https://github.com/devth/yetibot/blob/db13cd00f3f84e0d69e40cf9e1144bb1e6d4263d/src/yetibot/commands/twitter.clj#L87

Something like:
```clojure
#"^show\s+(\S+)" show
```

That will send all `twitter show <username>` commands to a `show` function, which will look something like:

```clojure
(defn show
  "twitter show <screen-name> # show the top 10 tweets from <screen-name>"
  [{[_ screen-name] :match}]
    ;; TODO, use the twitter `model` to do the search
  )
```

Command handlers (such as `show` and `lookup`) get a map of args including things like the matched regex, raw args, user, etc. We're only interested in `:match` here - it contains the result of applying `re-find` to your regex and the string. For example:

```clojure
yetibot.core.repl=> (re-find #"show\s+(\S+)" "show yetibot")
["show yetibot" "yetibot"]
```

Since the first item in the vector is the full string, we ignore it and destructure the 2nd item into `screen-name`: `{[_ screen-name] :match}`.

Ping me if you need to!

**Comment by dev-hartmann** [comment]
Thanks for providing the awesome hints! Will start working on it this evening after work.

**Comment by manalipankaj** [comment]
Can we still contribute?

**Comment by dev-hartmann** [comment]
hey, i got a basic implementation done. i will push to my fork this evening after work. maybe you want to have a look and help me get this done? i'm still new to clojure, so there will be enough space for improvements.

**Comment by dev-hartmann** [comment]
I had the repo on my work machine, so I could send a PR. Please have a look at the code

https://github.com/devth/yetibot/pull/569

**Comment by devth** [comment]
@dev-hartmann responded on your PR.

@manalipankaj yep, you're welcome to contribute!

**Comment by devth** [comment]
Fixed by #569
